alert/store.go

```diff
@@ -791,3 +791,41 @@ func (s *Store) State(ctx context.Context, alertIDs []int) ([]State, error) {
 
 	return list, nil
 }
+
+func (s *Store) Feedback(ctx context.Context, alertID int) (f Feedback, err error) {
+	row, err := gadb.New(s.db).AlertFeedback(ctx, int64(alertID))
+	if errors.Is(err, sql.ErrNoRows) {
+		return Feedback{
+			AlertID:     alertID,
+			NoiseReason: "",
+		}, nil
+	}
+
+	return Feedback{
+		AlertID:     int(row.AlertID),
+		NoiseReason: row.NoiseReason.String,
+	}, err
+}
+
+func (s Store) UpdateFeedback(ctx context.Context, feedback *Feedback) error {
+	err := permission.LimitCheckAny(ctx, permission.System, permission.User)
+	if err != nil {
+		return err
+	}
+
+	f, err := (s).Feedback(ctx, feedback.AlertID)
+	if err != nil {
+		return err
+	}
+	f.NoiseReason = feedback.NoiseReason
+
+	err = gadb.New(s.db).SetAlertFeedback(ctx, gadb.SetAlertFeedbackParams{
```

Comment:
We need to validate the content and length of `NoiseReason` before passing it to the DB (e.g., `validate.Text`).